apache/incubator-zeppelin
# Shell interpreter improvements

### What is this PR for?
*Provide ability to to run  shell commands in parallel
*Provide ability to cancel shell command
*Propagate the error from shell commands to UI


### What type of PR is it?
Improvement

### Todos
NA

### Is there a relevant Jira issue?
No

### How should this be tested?
*To check parallelism, run more than 10 shell commands concurrently.
*To verify whether error is propagate to UI, execute a shell command which will error out(simplest being cd of a non existent directory )
*To verify the cancel functionality, try cancelling a shell command that is running.

### Screenshots (if appropriate)
NA

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No
